datapackage_pipelines_ckan/processors/dump/to_ckan.py

```diff
@@ -0,0 +1,222 @@
+import os
+import json
+import hashlib
+
+import datapackage as datapackage_lib
+from ckan_datapackage_tools import converter
+from datapackage_pipelines.lib.dump.dumper_base import FileDumper, DumperBase
+
+from datapackage_pipelines_ckan.utils import make_ckan_request
+
+import logging
+log = logging.getLogger(__name__)
+
+
+class CkanDumper(FileDumper):
+
+    def initialize(self, parameters):
+        super(CkanDumper, self).initialize(parameters)
+        self.ckan_base_url = '{ckan_host}/api/3/action/'.format(
+            ckan_host=parameters['ckan-host'])
+        self.ckan_api_key = parameters.get('ckan-api-key')
+        self.dataset_resources = []
+        self.dataset_id = None
+
+    def handle_resources(self, datapackage,
```

Comment:
There's quite a bit of repetition between this method and how it's defined in the `DumperBase` base class (see https://github.com/frictionlessdata/datapackage-pipelines/blob/307590fa908743e4b9c3de77a287fb332b03b0cf/datapackage_pipelines/lib/dump/dumper_base.py#L129-L150). I'm imagining you aren't reusing it either because it's re-defined in `FileDumper`, or because you need to execute the code in a different order. Is that right? Just checking in case there's an opportunity to DRY the code a bit.

Reply:
Yeah, I needed to change the order. `handle_resources` dictates the main flow of the class, and I need datapackages to be determined first.